modules/acquia_cms_search/acquia_cms_search.module

```diff
@@ -161,3 +162,80 @@ function acquia_cms_search_search_api_server_update(EntityInterface $entity) {
 function acquia_cms_search_form_acquia_cms_solr_search_form_alter(array &$form) {
   $form['#submit'][] = AcquiaSearchFacade::class . '::submitSettingsForm';
 }
+
+/**
+ * Implements hook_modules_installed().
+ */
+function acquia_cms_search_modules_installed(array $modules) {
+  $module_handler = Drupal::moduleHandler();
+  // Category facet is created if content type modules are installed.
+  if ($module_handler->moduleExists('acquia_cms_person') ||
+  $module_handler->moduleExists('acquia_cms_place') ||
+  $module_handler->moduleExists('acquia_cms_article') ||
+  $module_handler->moduleExists('acquia_cms_event') ||
+  $module_handler->moduleExists('acquia_cms_page')) {
+    // Load the category facet if it exists.
+    $category_facet = Facet::load('search_category');
+    // Create a new facet if the category facet doesn't exists.
+    if (empty($category_facet)) {
+      // Programmatically creating the facet.
+      $facet = Facet::create([
+        'id' => 'search_category',
+        'name' => 'Category',
+        'url_alias' => 'categroy',
+        'field_identifier' => 'field_categories',
+      ]
+      );
+      // Set category widget & related settings.
+      $facet->setWidget('links', [
+        'show_numbers' => TRUE,
+        'soft_limit' => 0,
+        'soft_limit_settings' => [
+          'show_less_label' => 'Show less',
+          'show_more_label' => 'Show more',
+        ],
+        'show_reset_links' => 'false',
+        'reset_text' => 'Show all',
+        'hide_reset_when_no_selection' => 0,
+      ]);
+      $facet->setWeight(0);
+
+      // Set default empty behaviour & other settings.
+      $facet->setEmptyBehavior(['behavior' => 'none']);
+      $facet->setThirdPartySetting('facets_pretty_paths', 'coder', 'taxonomy_term_coder');
```

Comment:
@panshulK We need to add this plugin only if module `facets_pretty_paths` is enabled. Please add that condition.

Reply:
Done